Content.Server/Weapon/Melee/Esword/EswordSystem.cs

```diff
@@ -0,0 +1,135 @@
+            SubscribeLocalEvent<EswordComponent, InteractUsingEvent>(OnInteractUsing);
+        }
+
+        private void OnComponentInit(EntityUid uid, EswordComponent comp, ComponentInit args)
+        {
+            string[] possibleColors = { "Tomato", "DodgerBlue", "Aqua", "MediumSpringGreen", "MediumOrchid" };
+            Random random = new Random();
+            comp.BladeColor = Color.FromName(possibleColors[random.Next(5)]);
+        }
+
+        private void OnMeleeHit(EntityUid uid, EswordComponent comp, MeleeHitEvent args)
+        {
+            
+            if (comp.Activated == true)
+            {
+                args.BonusDamage.DamageDict = new Dictionary<string, FixedPoint2>()
+                {
+                    {"Slash", FixedPoint2.New(25)},
+                };
+                args.HitSoundOverride = comp.HitSound;
+            }
+        }
+
+        private void OnUseInHand(EntityUid uid, EswordComponent comp, UseInHandEvent args)
+        {
+            if (!_blockerSystem.CanUse(args.User))
+                return;
+
+            if (comp.Activated)
+            {
+                TurnOff(comp);
+            }
+            else
+            {
+                TurnOn(comp, args.User);
+            }
+        }
+
+        private void TurnOff(EswordComponent comp)
+        {
+            if (!comp.Activated)
+                return;
+
+            if (!TryComp(comp.Owner, out SpriteComponent? sprite)
+                || !TryComp(comp.Owner, out SharedItemComponent? item))
+                return;
+
+            item.Size = 5;
+
+            SoundSystem.Play(Filter.Pvs(comp.Owner), comp.DeActivateSound.GetSound(), comp.Owner);
+
+            item.EquippedPrefix = "off";
+            sprite.LayerSetVisible(1, false);
+
+            comp.Activated = false;
+        }
+
+        private void TurnOn(EswordComponent comp, EntityUid user)
+        {
+            if (comp.Activated)
+                return;
+
+            if (!TryComp(comp.Owner, out SpriteComponent? sprite)
+                || !TryComp(comp.Owner, out SharedItemComponent? item))
+                return;
+
+            item.Size = 9999;
+
+            SoundSystem.Play(Filter.Pvs(comp.Owner), comp.ActivateSound.GetSound(), comp.Owner);
+
+            if (comp.Hacked == true)
+            {
+                item.EquippedPrefix = "on-rainbow";
+                sprite.LayerSetState(0, "e_sword_rainbow_on");
+            }
+            else
+            {
+                item.EquippedPrefix = "on";
+                item.Color = comp.BladeColor;
+
+                sprite.LayerSetColor(1, comp.BladeColor);
+                sprite.LayerSetVisible(1, true);
+            }
+            
+            comp.Activated = true;
+        }
+
+        private void OnInteractUsing(EntityUid uid, EswordComponent comp, InteractUsingEvent args)
+        {
+            if (_blockerSystem.CanInteract(args.User) || comp.Hacked == true)
```

Comment:
you found one of the missing `!`, but forgot this one.

Also tiny nitpick, but in general its better to reverse the order of operations here:
`comp.Hacked == true || _blockerSystem.CanInteract(args.User)`
because one is just an equality check, but the `Caninteract` is more costly, and there's no need to do it if the second check is true anyways.